numba_dpex/tests/core/types/test_box_unbox.py

```diff
@@ -0,0 +1,145 @@
+# SPDX-FileCopyrightText: 2020 - 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Tests for boxing and unboxing of types supported inside dpjit
+"""
+
+import functools
+import sys
+
+import dpctl
+import dpnp
+import numpy
+import pytest
+
+import numba_dpex as dpx
+
+constructors_with_parent = [
+    dpctl.SyclEvent,
+    dpctl.SyclQueue,
+    # numpy test is actually not related to the dpex, but helps make sure we are
+    # doing it in the right way. Testing tests =)
+    lambda: numpy.empty(10, dtype=numpy.float32),
+    lambda: dpnp.empty(10, dtype=dpnp.float32),
+    # TODO: is it possible to test USMNd array same way?
+    # lambda: dpx.USMNdArray(1, queue=None, dtype=dpx.float32),
+]
+
+ranges = [(10,), (10, 10), (10, 10, 10)]
+
+constructors_without_parent = [
+    functools.partial(dpx.Range, *r) for r in ranges
+] + [
+    functools.partial(dpx.NdRange, dpx.Range(*r), dpx.Range(*r)) for r in ranges
+]
+
+
+@pytest.fixture(params=constructors_with_parent + constructors_without_parent)
+def constructor(request):
+    return request.param
+
+
+@pytest.fixture(params=constructors_with_parent)
+def constructor_wp(request):
+    return request.param
+
+
+@pytest.fixture(params=constructors_without_parent)
+def constructor_np(request):
+    return request.param
+
+
+@dpx.dpjit
+def unbox_box(a):
+    return a
+
+
+# @numba.njit
```

Comment:
please remove commented out code. Also `dpjit` is for now equivalent to `njit(Parallel=True)`

Reply:
Ops, did not see the comment. Let me address this issue in separate PR.